src/Nri/Ui/Carousel/V2.elm

```diff
@@ -0,0 +1,206 @@
+module Nri.Ui.Carousel.V2 exposing
+    ( viewWithCombinedControls, viewWithPreviousAndNextControls
+    , viewWithTabControls, LabelledBy(..), Role(..)
+    )
+
+{-| Patch changes:
+
+  - added new carousel APIs (with tabbed controls/previous and next controls, and combined controls) using [W3C guidelines](https://www.w3.org/WAI/ARIA/apg/patterns/carousel/)
+
+@docs viewWithCombinedControls, viewWithPreviousAndNextControls
+@docs viewWithTabControls, LabelledBy, Role
+
+-}
+
+import Accessibility.Styled.Aria as Aria
+import Css exposing (..)
+import Html.Styled as Html exposing (..)
+import Html.Styled.Attributes as Attrs exposing (css)
+import TabsInternal.V2 as TabsInternal
+
+
+{-| Type which represents the type of aria label which will be used
+`LabelledByIdOfVisibleLabel` will point to an existing element id on the DOM
+`LabelledByAccessibleLabelOnly` will be a label of the element
+-}
+type LabelledBy
+    = LabelledByIdOfVisibleLabel String
+    | LabelledByAccessibleLabelOnly String
+
+
+{-| `Role`, which can be either [Group](https://w3c.github.io/aria/#group) or [Region](https://w3c.github.io/aria/#region)
+-}
+type Role
+    = Group
+    | Region
+
+
+{-| Builds a carousel with previous and next controls
+Returns:
+`controls`: previous and next buttons element
+`slides` the container with the carousel contents
+`containerAttributes` attributes that should be used on the parent div of both the button and slides elements
+-}
+viewWithPreviousAndNextControls :
+    { selected : id
+    , panels :
+        List
+            { id : id
+            , slideHtml : Html msg
+            , labelledBy : LabelledBy
+            }
+    , viewPreviousButton : Html msg
+    , viewNextButton : Html msg
+    , labelledBy : LabelledBy
+    , role : Role
+    }
+    ->
+        { viewPreviousButton : Html msg
+        , viewNextButton : Html msg
+        , slides : Html msg
+        , containerAttributes : List (Attribute msg)
+        }
+viewWithPreviousAndNextControls config =
+    { viewPreviousButton = config.viewPreviousButton
+    , viewNextButton = config.viewNextButton
+    , slides =
+        List.map
+            (\panel ->
+                Html.div
+                    [ Attrs.attribute "role" "group"
+                    , Aria.roleDescription "slide"
+                    , labelledByToAttr panel.labelledBy
+                    , css
+                        [ if config.selected == panel.id then
+                            Css.display Css.block
+
+                          else
+                            Css.display Css.none
+                        ]
+                    ]
+                    [ panel.slideHtml ]
+            )
+            config.panels
+            |> Html.div [ Attrs.attribute "atomic" "false", Attrs.attribute "live" "polite" ]
```

Comment:
I know the authoring practices guide recommends putting the slides in a live region, but I think we want to instead manage announcements ourselves more directly. Using live regions, you can end with up content getting way over-announced, and it's difficult to fix once it starts happening. 

For example, go to a regular Guided Draft, turn on VoiceOver, and click into a component -- because the sidebar content changes, the tips and tricks carousel that currently uses aria-live gets announced in a confusing and unhelpful way.

Maybe we can expose a standard announcement from this component for when the selected slide changes, and document that it should be used when the slide changes? Or maybe adjust the api such that it takes a field like `selectAndAnnounce : { select : id, announce : String } -> msg`?

Reply:
I don't know if I quite get what this selectAndAnnouce would do, could you give me an example for it?

Reply:
GitHub isn't letting me upload a video right now, annoying!

There are basically 2 ways of communicating to SR users that something has changed elsewhere on the page:
- move focus to the changed item (for example, when navigating between SPA pages, we move focus to the new page's heading)
- use a live region

For our purposes, we _only_ want our AT announcements center to include live regions. Everywhere else, we don't want live regions. So anything that would _otherwise_ be something we'd use a live region to communicate, we instead want to send a (potentially shortened) message to the ATAC instead (with Nri.Effect.announcePolitely and/or Nri.Effect.announceAssertively).

I do see that you're currently trying to move focus to the newly-selected slide. This works in theory (although I'm not sure if it's ergonomic for keyboard users?). However, for the slide to actually _accept_ focus, you'll need to add `Accessibility.Styled.Key.tabbable False` (which adds tabindex=-1).